File: Legacy/ERP.Entities/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ERP.Entities
{
    public class Category : BaseClass
    {
        public int CatId { get; set; }
        public string CatName { get; set; }
        public string CatSubName1 { get; set; }
        public string CatSubName2 { get; set; }
        public int CatOrder { get; set; }
        public bool CatIsActived { get; set; }
        public string CatImagePath { get; set; }
        public bool CatDisplayInMenu { get; set; }
        public bool CatDisplayInExhibition { get; set; }
        public int? CatParentCatId { get; set; }
        public string CatParentCatName { get; set; }
        public string CatDescription { get; set; }
        public int SocId { get; set; }

        public int PrdCount { get; set; }

        /// <summary>
        /// only for display
        /// </summary>
        public int CatLevel { get; set; }

        public IEnumerable<Category> SubCategories { get; set; }
    }
}

File: frontend/src/types/category.ts
export interface Category {
  cat_id: number
  cat_name: string
  cat_sub_name_1?: string | null
  cat_sub_name_2?: string | null
  cat_order?: number
  cat_is_actived?: boolean
  cat_image_path?: string | null
  cat_display_in_menu?: boolean
  cat_display_in_exhibition?: boolean
  cat_parent_cat_id?: number | null
  soc_id: number
  cat_description?: string | null
}

export interface CategoryListResponse {
  items: Category[]
  total: number
  skip: number
  limit: number
}

export interface CategoryCreateDto {
  name: string
  subName1?: string
  subName2?: string
  order?: number
  isActive?: boolean
  imagePath?: string
  displayInMenu?: boolean
  displayInExhibition?: boolean
  parentId?: number
  societyId?: number
  description?: string
}

export interface CategoryUpdateDto extends Partial<CategoryCreateDto> {}

